src/main/java/com/welovecoding/nbeditorconfig/io/reader/FileInfoReader.java

```diff
@@ -163,9 +164,53 @@ private static String readFirstLineWithSeparator(FileObject fo, Charset charset)
     return firstLine;
   }
 
-  public static String trimTrailingWhitespace(Stream<String> lines, String lineEnding) {
-    return lines.map((String content) -> {
-      return content.replaceAll("\\s+$", "");
-    }).collect(Collectors.joining(lineEnding));
+  public static String trimTrailingWhitespace(Collection<String> lines, String lineEnding) {
+    StringBuilder sb = new StringBuilder();
+    for (String content : lines) {
+      sb.append(content.replaceAll("\\s+$", ""));
+      sb.append(lineEnding);
+    }
+    return sb.toString().trim();
+  }
+
+  public static String trimTrailingWhitespace(String text, String lineEnding) {
+    List<String> lines = new ArrayList<>();
+    {
+      BufferedReader reader = new BufferedReader(new StringReader(text));
+      try {
+        for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+          lines.add(line);
+        }
+        reader.close();
+      } catch (IOException ex) {
+        Exceptions.printStackTrace(ex);
+      }
+    }
+    return trimTrailingWhitespace(lines, lineEnding);
+  }
+
+  public static String replaceLineEndings(Collection<String> lines, String lineEnding) {
+    StringBuilder sb = new StringBuilder();
+    for (String content : lines) {
+      sb.append(content);
+      sb.append(lineEnding);
+    }
+    return sb.toString().trim();
+  }
+
+  public static String replaceLineEndings(String text, String lineEnding) {
+    List<String> lines = new ArrayList<>();
+    {
+      BufferedReader reader = new BufferedReader(new StringReader(text));
+      try {
+        for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+          lines.add(line);
+        }
+        reader.close();
```

Comment:
Shouldn't this `reader` be closed in a `finally` block or placed in a `try-with-resources` block?